### ISSUE TITLE
Add colorbar to 2D graphs

### ISSUE THREAD

**Issue by maffoo** [labrad/scalabrad-web#90]

**Comment by jwenner** [comment]
What's the status of this? Thanks.

**Comment by joshm-3b6b58** [comment]
It's not being worked on. Shouldn't too too difficult to add. What priority
would you rank this with the other issues listed?

**Comment by maffoo** [comment]
I think @ccwa is working on this, actually. Perhaps she can weigh in on the
current status.

**Comment by joshm-3b6b58** [comment]
Marking this as a bug since it's needed to reach feature parity.

**Comment by adamcw** [comment]
As I have never seen the Delphi grapher in action, can someone describe this feature so I can see how difficult it may be to implement?

**Comment by adamcw** [comment]
If @ccwa has made progress, it would be great to see as I will likely need to port it to use WebGL once my branch is merged into master.

**Comment by jwenner** [comment]
@adamcw, I've included a screenshot of a 2D plot from the Delphi grapher. The colorbar is shown to the right (with the tick marks for 100, 200, 300, 400, 500). This provides a way to associate colors on a 2D plot with the values of the dependent variable which is plotted.

In addition, note that the limits on the colorbar dynamically update as new data are added. In particular, as the limits of the dependent variable are widened due to data being outside the old limits, the colorbar ticks and coloring is updated accordingly.

I hope this helps. Let me know if you have more questions.

![image](https://cloud.githubusercontent.com/assets/8183919/16393268/ef32d570-3c64-11e6-8d29-841a7be19c5b.png)

**Comment by adamcw** [comment]
So it's essentially an axis for the third variable, the color?

**Comment by adamcw** [comment]
Turns out setting a base href on a page messes with SVG linking, and it took multiple hours to diagnose why the gradient on my SVG wasn't showing up...

This looks fairly straight forward though, at least in the realm of d3/SVG. I think this will be a good short term solution to get to feature parity sooner rather than later.
